src/lib/features/dependent-features/dependent-features-service.ts

```diff
@@ -148,82 +149,81 @@ export class DependentFeaturesService {
                       variants,
                   };
         await this.dependentFeaturesStore.upsert(featureDependency);
-        await this.eventService.storeEvent({
-            type: 'feature-dependency-added',
-            project: projectId,
-            featureName: child,
-            createdBy: user,
-            createdByUserId: userId,
-            data: {
-                feature: parent,
-                enabled: featureDependency.enabled,
-                ...(variants !== undefined && { variants }),
-            },
-        });
+        await this.eventService.storeEvent(
+            new FeatureDependencyAddedEvent({
+                project: projectId,
+                featureName: child,
+                auditUser,
+                data: {
+                    feature: parent,
+                    enabled: featureDependency.enabled,
+                    ...(variants !== undefined && { variants }),
+                },
+            }),
+        );
     }
 
     async deleteFeatureDependency(
         dependency: FeatureDependencyId,
         projectId: string,
         user: IUser,
+        auditUser: IAuditUser,
```

Comment:
For the future, we can consider merging `auditUser` with `user` as they should always represent the same, but user has permissions and auditUser don't, but for now, let's do this as a gradual improvement.

Reply:
yes, agreed, the thing missing was having some way from audit user to UserWithPermission. so the one thing audituser has that most other user implementations do not have is the ip.